kernel/sched.c

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <sched.h>
+#include <syscall.h>
+#include <unistd.h>
+
+#include <myst/eraise.h>
+#include <myst/errno.h>
+#include <myst/kernel.h>
+#include <myst/thread.h>
+
+long myst_syscall_sched_getparam(pid_t pid, struct sched_param* param)
+{
+    long ret = 0;
+    long params[6] = {0};
+
+    if (pid < 0)
+        ERAISE(-EINVAL);
+    else if (pid >= MYST_PID_MAX)
+        ERAISE(-ESRCH);
+
+    if (param && !myst_is_addr_within_kernel(param))
+    {
+        ERAISE(-EFAULT);
+    }
```

Comment:
if it exists and is valid memory please please memset here to zero

Reply:
I memset param to 0 here - https://github.com/RRathna/mystikos/blob/41d8e8da122231c9c50f3a4e1044de3e9f3e6ea9/crt/sched.c#L10-L11, should I move it here?

Reply:
I know! But the kernel trusts no one so needs to be done here.